### PR TITLE
Update HeanCMS theme

### LINKED ISSUE #1943: perf scan no pages found
### Source information

perf scan 1.4.20

### Source language

french

### Steps to reproduce

Read a chapter

### Expected behavior

chapter display

### Actual behavior

Error : no pages found

### Mihon/Tachiyomi version

tachiyomiSY 1.10.5

### Android version

Android 13

### Other details

_No response_

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] I have updated the app to version **[0.15.3](https://github.com/mihonapp/mihon/releases/latest)**.
- [X] I have updated all installed extensions.
- [X] I have tried the [troubleshooting guide](https://tachiyomi.org/help/guides/troubleshooting/).
- [ ] I have admitted that I am a clown by having checked this box, as I have not read these acknowledgements.
- [X] If this is an issue with the app itself, I should be opening an issue in the [app repository](https://github.com/tachiyomiorg/tachiyomi/issues/new/choose).
- [X] I will fill out all of the requested information in this form.

### LINKED ISSUE #1953: Omega Scans Extention not working
### Source information

Omega Scans

### Source language

English

### Steps to reproduce

When going to "browse" then "Source" then Omega scans its just empty + it doesnt update the manga in library
Screenshot_20240318_005140_Mihon

### Expected behavior

It should be full with manga / manwha and update the manga / manwha when refreshed

### Actual behavior

Its empty like shown in the picture and wont update manga / manwha in library
Screenshot_20240318_005140_Mihon

### Mihon/Tachiyomi version

0.16.4

### Android version

Android 14

### Other details

_No response_

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] I have updated the app to version **[0.15.3](https://github.com/mihonapp/mihon/releases/latest)**.
- [X] I have updated all installed extensions.
- [X] I have tried the [troubleshooting guide](https://tachiyomi.org/help/guides/troubleshooting/).
- [ ] I have admitted that I am a clown by having checked this box, as I have not read these acknowledgements.
- [X] If this is an issue with the app itself, I should be opening an issue in the [app repository](https://github.com/tachiyomiorg/tachiyomi/issues/new/choose).
- [X] I will fill out all of the requested information in this form.

### PR DESCRIPTION
**Pull request by bapeey**
OmegaScans users will have to migrate their series to OmegaScans
TempleScans users will have to migrate their series to TempleScans

- Removed the old and unused endpoints
- Fix PerfScan pageList (Closes #1943)
- FIx OmegaScans (Closes #1953)

Checklist:

- [ ] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [x] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [x] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [ ] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [x] Have not changed source names
- [ ] Have explicitly kept the `id` if a source's name or language were changed
- [x] Have tested the modifications by compiling and running the extension through Android Studio
- [ ] Have removed `web_hi_res_512.png` when adding a new extension

### CODE REVIEW

**Review by AwkwardPeak7 — COMMENTED** [review]
since none of the current sources use random urls, maybe remove that whole logic like slug in preference etc to simplify it.

**Comment by bapeey** [comment]
TempleScans users will have to migrate their series too

File: lib-multisrc/heancms/build.gradle.kts [hunk 1]
```diff
@@ -2,4 +2,8 @@ plugins {
     id("lib-multisrc")
 }
 
-baseVersionCode = 20
+baseVersionCode = 21
+
+dependencies {
+    api(project(":lib:i18n"))
+}
```
Comment (AwkwardPeak7) [inline lib-multisrc/heancms/build.gradle.kts]:
```suggestion
}

```
newline

File: lib-multisrc/heancms/src/eu/kanade/tachiyomi/multisrc/heancms/HeanCms.kt [hunk 2]
```diff
@@ -134,56 +91,19 @@ abstract class HeanCms(
         if (json.startsWith("{")) {
```
Comment (AwkwardPeak7) [inline lib-multisrc/heancms/src/eu/kanade/tachiyomi/multisrc/heancms/HeanCms.kt]: is the `querysearch` and not `querysearch` still relevant?

File: lib-multisrc/heancms/src/eu/kanade/tachiyomi/multisrc/heancms/HeanCms.kt [hunk 3]
```diff
@@ -292,223 +164,160 @@ abstract class HeanCms(
             .addQueryParameter("page", page.toString())
             .addQueryParameter("perPage", "12")
             .addQueryParameter("tags_ids", tagIds)
+            .addQueryParameter("adult", "true")
 
         return GET(url.build(), headers)
     }
 
     override fun searchMangaParse(response: Response): MangasPage {
```
Comment (AwkwardPeak7) [inline lib-multisrc/heancms/src/eu/kanade/tachiyomi/multisrc/heancms/HeanCms.kt:156]:
```
override fun searchMangaParse(response: Response) = popularMangaParse(response)
```
I dont see a difference now 🤔

File: lib-multisrc/heancms/src/eu/kanade/tachiyomi/multisrc/heancms/HeanCms.kt [hunk 4]
```diff
@@ -292,223 +164,160 @@ abstract class HeanCms(
...
-        val chapterId = chapter.url.substringAfterLast("#").substringBefore("-paid")
+        val result = response.parseAs<HeanCmsSeriesDto>()
 
-        val apiHeaders = headersBuilder()
-            .add("Accept", ACCEPT_JSON)
-            .build()
+        val currentTimestamp = System.currentTimeMillis()
 
-        return GET("$apiUrl/series/chapter/$chapterId", apiHeaders)
+        val showPaidChapters = preferences.showPaidChapters
```
Comment (AwkwardPeak7) [inline lib-multisrc/heancms/src/eu/kanade/tachiyomi/multisrc/heancms/HeanCms.kt]: move this to the top and use it in the other endpoint parsing as well

**Review by AwkwardPeak7 — CHANGES_REQUESTED** [review]
also should remove existing slug_map from preference on startup. 
I think only perfScan should have it (right?) so add this to it:
```
init {
        preferences.run {
            if (contains("pref_url_map")) {
                edit().remove("pref_url_map").apply()
            }
        }
    }
```

File: src/fr/perfscan/src/eu/kanade/tachiyomi/extension/fr/perfscan/PerfScan.kt [hunk 5]
```diff
@@ -1,17 +1,30 @@
...
 import eu.kanade.tachiyomi.network.interceptor.rateLimitHost
 import okhttp3.HttpUrl.Companion.toHttpUrl
 import okhttp3.OkHttpClient
+import uy.kohesive.injekt.Injekt
+import uy.kohesive.injekt.api.get
 
 class PerfScan : HeanCms("Perf Scan", "https://perf-scan.fr", "fr") {
+
+    private val preferences: SharedPreferences =
+        Injekt.get<Application>().getSharedPreferences("source_$id", 0x0000)
```
Comment (AwkwardPeak7) [inline src/fr/perfscan/src/eu/kanade/tachiyomi/extension/fr/perfscan/PerfScan.kt]: just make it protected in the hean class so it is available here

**Review by AwkwardPeak7 — APPROVED** [review]
poor iguanas

**Pull request merged**